### PR TITLE
p2p: when peer is removed remove it also from dial history

### PR DESCRIPTION
**Pull request by dshulyak**
This change removes a peer information from dialing history when a peer is removed from a static list.
It allows forcing a server to re-dial concrete peer if it is needed.

In our case, we are running geth node on mobile devices. It is common
for a network connection to flap on mobile. Almost every time it flaps
or network connection is changed from cellular to wifi peers are disconnected
with read timeout. And usually, it takes 30 seconds (default expiration timeout)
to recover connection with static peers after connectivity is restored.

This change allows us to reconnect with peers almost immediately and it seems harmless enough.

### CODE REVIEW

**Comment by dshulyak** [comment]
@fjl @zsfelfoldi any chance to land it before 1.8.1 cut off?

File: p2p/dial.go [hunk 1]
```diff
@@ -191,6 +194,7 @@ func (s *dialstate) newTasks(nRunning int, peers map[discover.NodeID]*Peer, now
 	// Create dials for static nodes if they are not connected.
 	for id, t := range s.static {
 		err := s.checkDial(t.dest, peers)
+		fmt.Println("tests", "reason", err)
```
Comment (holiman) [inline p2p/dial.go]: Should use the logging facilities here

Comment (dshulyak) [reply p2p/dial.go]: sorry, that's an unfortunate leftover

**Comment by fjl** [comment]
The purpose of dial history is to avoid hammering the static node with connection requests. Please keep that in mind.

**Pull request merged**